rpcs3/rpcs3qt/gamepads_settings_dialog.cpp

```diff
@@ -0,0 +1,219 @@
+#include "gamepads_settings_dialog.h"
+#include "../Emu/Io/PadHandler.h"
+#include "../ds4_pad_handler.h"
+#include "../xinput_pad_handler.h"
+#include "../mm_joystick_handler.h"
+#include "../keyboard_pad_handler.h"
+#include "../Emu/Io/Null/NullPadHandler.h"
+//#include "emu_settings.h"
+#include "../Emu/System.h"
+
+input_config input_cfg;
+
+gamepads_settings_dialog::gamepads_settings_dialog(QWidget* parent)
+	: QDialog(parent)
+{
+	setWindowTitle(tr("Gamepads Settings"));
+
+	QVBoxLayout *dialog_layout = new QVBoxLayout();
+	QHBoxLayout *all_players = new QHBoxLayout();
+
+	input_cfg.load();
+
+	auto fill_device_combo = [](QComboBox *combo)
+	{
+		std::vector<std::string> str_inputs = input_cfg.player_input[0].to_list();
+		for (int index = 0; index < str_inputs.size(); index++)
+		{
+			combo->addItem(str_inputs[index].c_str());
+		}
+	};
+
+	for (int i = 0; i < 7; i++)
+	{
+		QGroupBox *grp_player = new QGroupBox(QString(tr("Player %1").arg(i+1)));
+
+		QVBoxLayout *ppad_layout = new QVBoxLayout();
+
+		co_inputtype[i] = new QComboBox();
+		fill_device_combo(co_inputtype[i]);
+		ppad_layout->addWidget(co_inputtype[i]);
+
+		co_deviceID[i] = new QComboBox();
+		co_deviceID[i]->setEnabled(false);
+		ppad_layout->addWidget(co_deviceID[i]);
+
+		QHBoxLayout *button_layout = new QHBoxLayout();
+		bu_config[i] = new QPushButton("Config");
+		bu_config[i]->setEnabled(false);
+		button_layout->addSpacing(bu_config[i]->sizeHint().width()*0.50f);
+		button_layout->addWidget(bu_config[i]);
+		button_layout->addSpacing(bu_config[i]->sizeHint().width()*0.50f);
+		ppad_layout->addLayout(button_layout);
+
+		grp_player->setLayout(ppad_layout);
+		all_players->addWidget(grp_player);
+
+		connect(co_inputtype[i], &QComboBox::currentTextChanged, [=] { ChangeInputType(i); });
+		connect(co_deviceID[i], &QComboBox::currentTextChanged, [=] { ChangeDevice(i); });
+		connect(bu_config[i], &QAbstractButton::clicked, [=] { ClickConfigButton(i); });
+
+		if (i == 3)
+		{
+			dialog_layout->addLayout(all_players);
+			all_players = new QHBoxLayout();
+			all_players->addStretch();
+		}
+	}
+
+	all_players->addStretch();
+	dialog_layout->addLayout(all_players);
+
+	QHBoxLayout *buttons_layout = new QHBoxLayout();
+	QPushButton *ok_button = new QPushButton("OK");
```

Comment:
Btw, isn't it better done via http://doc.qt.io/qt-5.9/qdialogbuttonbox.html#details ?